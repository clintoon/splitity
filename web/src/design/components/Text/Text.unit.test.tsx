import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import {
  Text,
  TEXT_TEST_ID,
  TextStyle,
  TextAs,
  TextAlign,
} from '@web/design/components/Text/Text';

const TEXT_CHILD = 'I am some words';

interface RenderTextOptions {
  styleOf?: TextStyle;
  as?: TextAs;
  margin?: string;
  textAlign?: TextAlign;
}

const renderText = (options: RenderTextOptions): RenderResult => {
  return render(<Text {...options}>{TEXT_CHILD}</Text>);
};

describe('<Text/>', (): void => {
  describe('test html tag', (): void => {
    it('displays correct html tag for default props', (): void => {
      const { getByTestId } = renderText({});
      expect(getByTestId(TEXT_TEST_ID)).toContainHTML('div');
    });

    it('displays correct html tag for when as prop is div', (): void => {
      const { getByTestId } = renderText({ as: TextAs.Div });
      expect(getByTestId(TEXT_TEST_ID)).toContainHTML('div');
    });

    it('displays correct html tag for when as prop is h1', (): void => {
      const { getByTestId } = renderText({ as: TextAs.H1 });
      expect(getByTestId(TEXT_TEST_ID)).toContainHTML('h1');
    });

    it('displays correct html tag for when as prop is h2', (): void => {
      const { getByTestId } = renderText({ as: TextAs.H2 });
      expect(getByTestId(TEXT_TEST_ID)).toContainHTML('h2');
    });

    it('displays correct html tag for when as prop is h3', (): void => {
      const { getByTestId } = renderText({ as: TextAs.H3 });
      expect(getByTestId(TEXT_TEST_ID)).toContainHTML('h3');
    });

    it('displays correct html tag for when as prop is h4', (): void => {
      const { getByTestId } = renderText({ as: TextAs.H4 });
      expect(getByTestId(TEXT_TEST_ID)).toContainHTML('h4');
    });

    it('displays correct html tag for when as prop is h5', (): void => {
      const { getByTestId } = renderText({ as: TextAs.H5 });
      expect(getByTestId(TEXT_TEST_ID)).toContainHTML('h5');
    });

    it('displays correct html tag for when as prop is h6', (): void => {
      const { getByTestId } = renderText({ as: TextAs.H6 });
      expect(getByTestId(TEXT_TEST_ID)).toContainHTML('h6');
    });

    it('displays correct html tag for when as prop is p', (): void => {
      const { getByTestId } = renderText({ as: TextAs.P });
      expect(getByTestId(TEXT_TEST_ID)).toContainHTML('p');
    });
  });

  describe('test rendering of text', (): void => {
    describe('as prop', (): void => {
      it('displays text when as prop is h1', (): void => {
        const { getByTestId } = renderText({ as: TextAs.H1 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when as prop is h2', (): void => {
        const { getByTestId } = renderText({ as: TextAs.H2 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when as prop is h3', (): void => {
        const { getByTestId } = renderText({ as: TextAs.H3 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when as prop is h4', (): void => {
        const { getByTestId } = renderText({ as: TextAs.H4 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when as prop is h5', (): void => {
        const { getByTestId } = renderText({ as: TextAs.H5 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when as prop is h6', (): void => {
        const { getByTestId } = renderText({ as: TextAs.H6 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when as prop is div', (): void => {
        const { getByTestId } = renderText({ as: TextAs.Div });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when as prop is p', (): void => {
        const { getByTestId } = renderText({ as: TextAs.P });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });
    });

    describe('styleOf prop', (): void => {
      it('displays text when styleOf prop is body', (): void => {
        const { getByTestId } = renderText({ styleOf: TextStyle.Body });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when styleOf prop is title1', (): void => {
        const { getByTestId } = renderText({ styleOf: TextStyle.Title1 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when styleOf prop is title2', (): void => {
        const { getByTestId } = renderText({ styleOf: TextStyle.Title2 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when styleOf prop is title3', (): void => {
        const { getByTestId } = renderText({ styleOf: TextStyle.Title3 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when styleOf prop is title4', (): void => {
        const { getByTestId } = renderText({ styleOf: TextStyle.Title4 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when styleOf prop is title5', (): void => {
        const { getByTestId } = renderText({ styleOf: TextStyle.Title5 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when styleOf prop is title6', (): void => {
        const { getByTestId } = renderText({ styleOf: TextStyle.Title6 });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });
    });

    it('displays text when there is margin', (): void => {
      const { getByTestId } = renderText({ margin: '4 4 4 4' });
      expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
    });

    describe('textAlign', (): void => {
      it('displays text when text align is center', (): void => {
        const { getByTestId } = renderText({ textAlign: TextAlign.Center });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when text align is right', (): void => {
        const { getByTestId } = renderText({ textAlign: TextAlign.Right });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });

      it('displays text when text align is left', (): void => {
        const { getByTestId } = renderText({ textAlign: TextAlign.Left });
        expect(getByTestId(TEXT_TEST_ID)).toHaveTextContent(TEXT_CHILD);
      });
    });
  });
});
