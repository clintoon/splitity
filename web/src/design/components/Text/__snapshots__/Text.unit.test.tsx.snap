// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> as prop has correct html tag for div 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> as prop has correct html tag for h1 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
}

<h1
  class="c0"
>
  example
</h1>
`;

exports[`<Text/> as prop has correct html tag for h2 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
}

<h2
  class="c0"
>
  example
</h2>
`;

exports[`<Text/> as prop has correct html tag for h3 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
}

<h3
  class="c0"
>
  example
</h3>
`;

exports[`<Text/> as prop has correct html tag for h4 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
}

<h4
  class="c0"
>
  example
</h4>
`;

exports[`<Text/> as prop has correct html tag for h5 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
}

<h5
  class="c0"
>
  example
</h5>
`;

exports[`<Text/> as prop has correct html tag for h6 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
}

<h6
  class="c0"
>
  example
</h6>
`;

exports[`<Text/> as prop has correct html tag for p 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
}

<p
  class="c0"
>
  example
</p>
`;

exports[`<Text/> has correct default html tag and css properties 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> margin prop has correct CSS properties when set 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 1 1 1 1;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> styleOf prop has correct CSS properties for body 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> styleOf prop has correct CSS properties for title1 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 64px;
  margin: 0;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> styleOf prop has correct CSS properties for title2 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 32px;
  margin: 0;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> styleOf prop has correct CSS properties for title3 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 28px;
  margin: 0;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> styleOf prop has correct CSS properties for title4 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 24px;
  margin: 0;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> styleOf prop has correct CSS properties for title5 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 20px;
  margin: 0;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> styleOf prop has correct CSS properties for title6 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 16px;
  margin: 0;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> textAlign prop has correct CSS properties for center 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
  text-align: center;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> textAlign prop has correct CSS properties for left 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
  text-align: left;
}

<div
  class="c0"
>
  example
</div>
`;

exports[`<Text/> textAlign prop has correct CSS properties for right 1`] = `
.c0 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  margin: 0;
  text-align: right;
}

<div
  class="c0"
>
  example
</div>
`;
